projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-detail-approval-details/order-detail-approval-details.component.ts

```diff
@@ -0,0 +1,15 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { Order } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { OrderDetailsService } from '../order-details.service';
+
+@Component({
+  selector: 'cx-order-details-approval-details',
+  templateUrl: './order-detail-approval-details.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OrderDetailApprovalDetailsComponent {
```

Comment:
Architecture smell:
- order approval component lives in the storefront lib
- order approval facade and ngrx state live in the my account lib

Reply:
yes, I think @marlass was mitigating this when keeping some order approval stuff in core.

Reply:
I don't think something like order approval state (for particular order) is in the my account lib.
This particular component enhances the order page and works only on the `order` data.

It is not related with my-company by code in any way.
However we might think about this component that it will be only present in storefronts that also use my-company. Then the my-account lib would be enabling support for this approval detail in the order page. With that thinking we would need to verify my assumption that order approval details should be used always with my-account lib.

With this component we should also think about `OrderDetailHeadlineComponent`, because even if that was already present in the storefrontlib now it is also aware of B2B Organization.

Idea for that: extend `OrderDetailHeadlineComponent` in `my-company` and in this case my-account component version would be used instead of the default.

Reply:
Agreed that order approval state doesn't belong here. We can certainly see customers managing the order approval without the my company self-service (i.e. integration from external system or in backoffice), which is why we should not depend it on my company.